keps/sig-testing/1746-stage-e2e-test-framework/README.md

```diff
@@ -0,0 +1,610 @@
+https://github.com/kubernetes/kubernetes/tree/master/staging.
+
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable, can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/20190731-production-readiness-review-process.md
+
+Production readiness review questionnaire must be completed for features in
+v1.19 or later, but is non-blocking at this time. That is, approval is not
+required in order to be in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+## Implementation History
+
+<!--
+Major milestones in the life cycle of a KEP should be tracked in this section.
+Major milestones might include
+- the `Summary` and `Motivation` sections being merged signaling SIG acceptance
+- the `Proposal` section being merged signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+Current work for decoupling the e2e test framework from core Kubernetes is bein
+tracked in this umbrella issue:
+*  https://github.com/kubernetes/kubernetes/issues/74352
+
+Work for cleaning up the e2e test framework can be found in this umbrella
+issue:
+* https://github.com/kubernetes/kubernetes/issues/76206
+
+Umbrella issue for tracking work aimed at decoupling the core functionality of
+the e2e test framework for e2e test configuration from utility functions can be
+found here
+* https://github.com/kubernetes/kubernetes/issues/81245
+
+## Alternatives
+
+<!--
+What other approaches did you consider and why did you rule them out?  These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+
+Another possibility is to create a new e2e test framework from scratch based on
+all the current requirements and infrastructure used in the Kubernetes community.
+However, this KEP is aimed at improving the current experience of Kubernetes maintainers.
+Our main concern is with facilitating the work without introducing new processes
+or requiring new infrastructure or components to be used.
+
+Although, if a new e2e test framework was to later be created, the current e2e
```

Comment:
couldn't the main project move to this new framework, once the framework was proven?

Reply:
I suppose.